### DISCUSSION TITLE
Sign and Submit Transaction

### DISCUSSION THREAD

**Discussion by aryaramadika** [rankanizer/polkadart#459] [Q&A] [answered]
Hello team, how do I execute a simple transfer and signAndSend transaction? I've followed the example, but I'm encountering this error: 'Unhandled Exception: Exception: {code: 1002, message: Verification Error: Runtime error: Execution failed: Execution aborted due to trap: wasm trap: wasm unreachable instruction executed' when I submit the extrinsic.

**Comment by leonardocustodio** [comment]
What network are you trying to send the extrinsic to? Also have you run polkadart_cli ?

**Reply by aryaramadika** [reply]
i'm using the Westend network and have already run the polkadart_cl

**Reply by leonardocustodio** [reply]
I wonder if it could be related to this new Signed Extension added: https://github.com/leonardocustodio/polkadart/issues/454
Are you able to try on polkadot to see if it works fine?

**Reply by aryaramadika** [reply]
okay thanks for response , i'll try it on polkadot

**Reply by aryaramadika** [reply]
<img width="1056" alt="Screenshot 2024-06-20 at 20 14 23" src="https://github.com/leonardocustodio/polkadart/assets/79522791/51689f22-095f-4030-b3d7-f69f2f84f987">

I think it works fine on Polkadot. I want to test token transfer, but I don't have any DOT tokens. Do you have any recommendations on what network I should use for testing ?

**Reply by leonardocustodio** [reply]
You can make this change in the lib that @clangenb is using, it should work on Westend:
https://github.com/leonardocustodio/polkadart/commit/bed30266786ad2d37b9bf3b6d282941743537003

**Reply by leonardocustodio** [reply]
If it is easier, feel free to send a PR, and I will publish a new version. I'm OOO and will not be able to do that right now.

**Reply by clangenb** [reply]
Westend and Kusama are already upgraded and require the new signed extension. Polkadot/Paseo do not need any changes as of now.

**Reply by aryaramadika** [reply]
So what should I do? Do I need to upgrade the Polkadart version? Because I still get the same error. @clangenb

**Reply by clangenb** [reply, marked as answer]
I have submitted an upstream PR that fixes your issue. So you can check out that PR, or if you use polkadart as a dependency you can switch to git dependencies (but beware, git dependencies in dart are a hassle, it might make sense to wait until a new version is published.) See #460

**Reply by aryaramadika** [reply]
thanks it works well in Paseo 🙏

**Reply by aryaramadika** [reply]
I want to ask a question, but it's off-topic and might seem dumb. What is the best way to get transaction history?

**Reply by leonardocustodio** [reply]
The change on https://github.com/leonardocustodio/polkadart/pull/460 has been published under `polkadart 0.4.5`.
About transaction history there is no way to do that @aryaramadika , you need an indexer for that.
Otherwise, you need to go through all blocks see all extrinsics, and find the ones that you want, which, is obviously, undoable for any realtime software.

**Reply by aryaramadika** [reply]
Oh, okay, thank you 🙏. Sorry, I want to ask one more question. How can I get the raw seed or private key from an account? Is it possible or not?

**Reply by leonardocustodio** [reply]
You can't get through `Keyring`, but you can get it from the algo keypair itself. Here is an example: https://github.com/leonardocustodio/polkadart/blob/main/packages/sr25519/example/example.dart
